src/components/TodoTempItem/TodoTempItem.tsx

```diff
@@ -0,0 +1,43 @@
+import { useContext } from 'react';
+import classnames from 'classnames';
+import { TodoTempContext } from '../../context/TodoTempContext';
+
+type Props = {
+  isActive: boolean;
+};
+
+export const TodoTempItem: React.FC<Props> = ({ isActive }) => {
```

Comment:
You can use your `TodoItem` component

Reply:
My TodoItem has so many other functions that compared to TodoTempItem will be redundant

Reply:
> My TodoItem has so many other functions that compared to TodoTempItem will be redundant

But, the design is the same, so if something changes you have to change two components instead `TodoItem`.
You can create a UI component `TodoItem` without any logic and use it in both cases

Reply:
TodoItem focuses on displaying individual todo with their specific details, while TodoTempItem is dedicated to indicating a loading state when data is still being processed. They serve different purposes and have different logic to fulfill those purposes.

Reply:
A detailed explanation of why creating two components with the same UI but different logic can lead to several issues and is generally not considered a good practice. Here are some reasons why it's a bad idea:

1) Code Duplication: By having two separate components with similar UI, you are duplicating code. This can lead to maintenance challenges because any updates or bug fixes to the UI will need to be applied to both components, increasing the chances of inconsistencies and errors.

2) Inconsistent Behavior: When components have different logic, it can result in inconsistent behavior for seemingly similar UI elements. Users may not understand why similar-looking elements behave differently, which can lead to confusion and a poor user experience.

3) Difficulty in Maintenance: As your application grows, maintaining multiple components with similar UI but different logic becomes increasingly complex. It can be challenging to keep track of changes and ensure that both components remain in sync with each other.

4) Scalability Issues: If you decide to introduce additional variations of the same UI with different logic in the future, you will end up creating more and more similar components, making your codebase harder to manage and extending development time.

5) Reduced Reusability: One of the fundamental principles of component-based UI development is reusability. When you have multiple components with distinct logic but the same UI, you lose the benefits of component reusability, which can save time and effort in the long run.

To address these issues, it's advisable to create a single representing component with props or configuration options that allow you to customize the behavior of the component. This approach promotes code reuse, maintainability, and consistency in your application.

In your case, you can create a single "TodoItem" component that accepts props or configuration options to handle both regular todos and temporary todos. This way, you can have a single, well-maintained component that can adapt to different scenarios, ensuring a cleaner and more maintainable codebase.